sdk/resourcemanager/Azure.ResourceManager.Core/src/Generated/ResourceGroupContainer.cs

```diff
@@ -310,6 +310,92 @@ async Task<Page<ResourceGroup>> NextPageFunc(string nextLink, int? pageSizeHint)
             return PageableHelpers.CreateAsyncEnumerable(FirstPageFunc, NextPageFunc);
         }
 
+        /// <summary>
+        /// List the resource groups for this subscription.
+        /// </summary>
+        /// <param name="filter"> The filter to apply on the operation.&lt;br&gt;&lt;br&gt;You can filter by tag names and values. For example, to filter for a tag name and value, use $filter=tagName eq &apos;tag1&apos; and tagValue eq &apos;Value1&apos;. </param>
+        /// <param name="top"> The number of results to return. If null is passed, returns all resource groups. </param>
+        /// <param name="cancellationToken"> A token to allow the caller to cancel the call to the service. The default value is <see cref="CancellationToken.None" />. </param>
+        /// <returns> A collection of resource operations that may take multiple service requests to iterate over. </returns>
+        [ForwardsClientCalls]
+        public virtual Pageable<ResourceGroup> List(string filter, int? top, CancellationToken cancellationToken = default)
```

Comment:
can we have one call the other instead of double implement, or just use default params and 1 method

Reply:
Updated!